src/components/single-video/SingleVideo.jsx

```diff
@@ -0,0 +1,149 @@
+import styles from "./SingleVideo.module.css";
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import { SingleVideoIFrame } from "./SingleVideoIFrame";
+import { Chip, NotFound, VideoList } from "components";
+
+export const SingleVideo = ({
+  singleVideo,
+  sidepanVideos,
+  isLiked,
+  setIsLiked,
+  status,
+  chipText,
+}) => {
+  return (
+    <>
+      {status === "error" ? (
+        <NotFound
+          documentTitle="Video Not Found"
+          textToShow="This video isn't available anymore"
+        />
+      ) : (
+        <div
+          className={classNames("flex", "m-auto", "p-3", styles.singleVideo)}
+        >
+          <div className={styles.iFrame_container}>
+            {status === "loading" ? (
+              <SingleVideoIFrame loading={true} />
+            ) : (
+              <SingleVideoIFrame
+                video={singleVideo}
+                isLiked={isLiked}
+                setIsLiked={setIsLiked}
+              />
+            )}
+          </div>
+
+          <div className="ml-2">
+            {status === "loading" ? (
+              <Chip loading={true} className="mb-2 mx-1" />
+            ) : (
+              <Chip
+                textToShow={chipText}
+                activeChip={chipText}
+                className="mb-2 mx-1"
+              />
+            )}
+
+            <VideoList videos={sidepanVideos} status={status} />
+          </div>
+        </div>
+      )}
+    </>
+  );
+};
+
+SingleVideo.propTypes = {
+  singleVideo: PropTypes.shape({
+    _id: PropTypes.string,
+    categoryName: PropTypes.string,
+    creator: PropTypes.string,
+    creatorLogo: PropTypes.shape({
+      altText: PropTypes.string,
+      url: PropTypes.string,
+    }),
+    isInWatchLater: PropTypes.bool,
+    isLiked: PropTypes.bool,
+    thumbnail: PropTypes.shape({
```

Comment:
It's great that you are using propTypes too. By looking at this it is easy for me to understand what this component supposed to have. I too have to read about this to implement in mine. Great work !!